### PR TITLE
Plugin: update calls to use sdf::Errors output 

### PR DESCRIPTION
**Pull request by marcoag**
Signed-off-by: Marco A. Gutierrez <marco@openrobotics.org>

# 🎉 New feature

Work towards https://github.com/gazebosim/sdformat/issues/820.

Depends on: https://github.com/gazebosim/sdformat/pull/1141.

## Summary
Adds missing `Errors` structure parameters in a few methods of the Plugin class.

## Test it

Using the Plugin class should report all errors through `sdf::Errors` if the `errors` parameter is used, it should print them otherwise.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/sdf/Plugin.hh [hunk 1]
```diff
@@ -107,6 +119,14 @@ namespace sdf
     /// \param[in] _elem Element to insert.
     public: void InsertContent(const sdf::ElementPtr _elem);
 
+    /// \brief Insert an element into the plugin content. This does not
+    /// modify the values in the sdf::ElementPtr returned by the `Element()`
+    /// function.
+    /// \param[in] _elem Element to insert.
+    /// \param[out] _errors Vector of errors.
+    public: void InsertContent(const sdf::ElementPtr _elem,
```
Comment (azeey) [inline include/sdf/Plugin.hh:127]: Can we make the output param come first for consistency?

File: include/sdf/Plugin.hh [hunk 2]
```diff
@@ -188,6 +227,16 @@ namespace sdf
...
+    /// \param[in] _filename Filename of the shared library associated with
+    /// this plugin.
+    /// \param[in] _name The name of the plugin.
+    /// \param[in] _xmlContent Optional XML content that will be stored in
+    /// this plugin.
+    private: void Init(sdf::Errors &_errors, const std::string &_filename,
+                    const std::string &_name, const std::string &_xmlContent);
+
     /// \brief Private data pointer.
     std::unique_ptr<sdf::PluginPrivate> dataPtr;
```
Comment (azeey) [inline include/sdf/Plugin.hh:241]: Apparently, the `dataPtr` was public before (yikes!), and now that there's a `private:` member right above, it's changing `dataPtr` to private as well, breaking ABI.

File: include/sdf/Plugin.hh (hunk 1 above)
Comment (marcoag) [reply include/sdf/Plugin.hh:127]: Sure, I also did it for the other case of `InsertContent`.

File: include/sdf/Plugin.hh (hunk 2 above)
Comment (marcoag) [reply include/sdf/Plugin.hh]: Damn, I made it public then. I will follow up with a PR to main and make sure `dataPtr`s are not public there.

**Pull request merged**